details/encryption.md

```diff
@@ -158,41 +158,46 @@ composed from:
 1. The fixed string "private-attribution", encoded in ASCII, terminated with a
    single zero-valued byte.
 
-2. The [ASCII serialization of the match key provider
-   origin](https://datatracker.ietf.org/doc/html/rfc6454#section-6.2),
-   terminated with a single zero-valued byte.
-
-3. The [ASCII serialization of the helper party
-   origin](https://datatracker.ietf.org/doc/html/rfc6454#section-6.2),
-   terminated with a single zero-valued byte.
-
-4. The [ASCII serialization of the registrable domain for the current
+2. The [ASCII serialization of the registrable domain for the current
    site](https://url.spec.whatwg.org/#host-registrable-domain) encoded in ASCII
    as a period-separated sequence of
    [A-](https://datatracker.ietf.org/doc/html/rfc5890#section-2.3.2.1) or
    [NR-LDH](https://datatracker.ietf.org/doc/html/rfc5890#section-2.3.2.2)
    labels (that is, the ASCII version of a domain name), terminated with a
    single zero-valued byte.
 
+3. The [ASCII serialization of the match key provider
+   origin](https://datatracker.ietf.org/doc/html/rfc6454#section-6.2),
+   terminated with a single zero-valued byte.
+
+4. The [ASCII serialization of the helper party
+   origin](https://datatracker.ietf.org/doc/html/rfc6454#section-6.2),
+   terminated with a single zero-valued byte.
+
 5. The single-byte key identifier from the key configuration for the helper
    party.
 
 6. The current epoch, encoded as an two-byte integer in network byte order.
 
+7. A single byte that indicates the event type, set to either 0 (for a source
+   event) or 1 (for a trigger event).
+
 This produces the following process in pseudocode:
 
 ```python
-def ipa_info(mkp_origin, helper_origin, site_origin, key_id, epoch):
+def ipa_info(site_registrable_domain, mkp_origin, helper_origin,
+             key_id, epoch, event_type):
     return concat(encode_str("private-attribution"),
+              encode(0, 1),
+              ascii_origin(site_registrable_domain),
               encode(0, 1),
               ascii_origin(mkp_origin),
               encode(0, 1),
               ascii_origin(helper_origin),
               encode(0, 1),
-              ascii_origin(site_registrable_domain),
-              encode(0, 1),
               encode(key_id, 1),
-              encode(epoch, 2))
+              encode(epoch, 2),
+              encode(event_type, 1))
```

Comment:
If we are sending an event_type here then do we still need secret shares of is_trigger_bit to be passed? Since now all helpers would know that in clear?

Reply:
That is absolutely right. The helper parties can share the is_trigger bit. No savings to be had from it being sparse though because we only single it at first